CMakeLists.txt

```diff
@@ -446,6 +446,11 @@ AppendToDefines(HAVE_DECL_ISNAN)
 CHECK_CXX_SYMBOL_EXISTS(signgam "cmath" HAVE_DECL_SIGNGAM)
 AppendToDefines(HAVE_DECL_SIGNGAM)
 
+CHECK_CXX_SYMBOL_EXISTS(fdopen "stdio.h" HAVE_FDOPEN)
+IF(HAVE_FDOPEN)
+    LIST(APPEND DEFINES HAVE_FDOPEN)
+ENDIF()
```

Comment:
What about `AppendToDefines(HAVE_FDOPEN)` instead of this if? It is more succint :)